router/usecase/pools/routable_cw_alloy_transmuter_pool.go

```
@@ -108,9 +129,10 @@ func (r *routableAlloyTransmuterPoolImpl) ChargeTakerFeeExactIn(tokenIn sdk.Coin
 }
 
 // ChargeTakerFeeExactOut implements domain.RoutablePool.
-// Returns tokenOutAmount and does not charge any fee for transmuter pools.
-func (r *routableAlloyTransmuterPoolImpl) ChargeTakerFeeExactOut(tokenOut sdk.Coin) (outAmountAfterFee sdk.Coin) {
-	return sdk.Coin{}
+// Returns tokenInAmount and does not charge any fee for transmuter pools.
+func (r *routableAlloyTransmuterPoolImpl) ChargeTakerFeeExactOut(tokenIn sdk.Coin) (inAmountAfterFee sdk.Coin) {
+	tokenInAfterTakerFee, _ := poolmanager.CalcTakerFeeExactOut(tokenIn, r.GetTakerFee())
```

what is the expected behaviour if error occur?

---

oh, method `poolmanager.CalcTakerFeeExactOut` does return actually two coins thus that blank identifier is used not to ignore an error but that second coin.